docs/configuration.md

```diff
@@ -477,6 +477,49 @@ Apart from these, the following properties are also available, and may be useful
     <code>spark.io.compression.codec</code>.
   </td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.encryption.enabled</code></td>
+  <td>false</td>
+  <td>
+    Enable shuffle file encryption.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.encryption.keySizeBits</code></td>
+  <td>128</td>
+  <td>
+    Shuffle file encryption key size in bits. The valid number includes 128, 192 and 256.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.encryption.keygen.algorithm</code></td>
+  <td>HmacSHA1</td>
+  <td>
+    The algorithm to generate the key used by shuffle file encryption. The supported algorithms are
+    described in the KeyGenerator section of the Java Cryptography Architecture Standard Algorithm
+    Name Documentation,
+  </td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.chimera.crypto.cipher.suite</code></td>
+  <td>AES/CTR/NoPadding</td>
+  <td>
+    Cipher suite for shuffle file encryption. The cipher suite name is identical to the
+    transformations described in the Cipher section of the Java Cryptography Architecture
+    Standard Algorithm Name Documentation. Currently only "AES/CTR/NoPadding" algorithm is
+    supported.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.chimera.crypto.codec.classes.aes.ctr.nopadding</code></td>
+  <td>com.intel.chimera.codec.OpensslAesCtrCryptoCodec,com.intel.chimera.codec
+  .JceAesCtrCryptoCodec</td>
```

Comment:
Kinda weird to break the line in the middle of the class name; it's fine to have long lines in docs.